packages/zcli-apps/tests/functional/package.test.ts

```diff
@@ -36,3 +39,53 @@ describe('package', function () {
     .catch(err => expect(err.message).to.contain('Error: invalid location'))
     .it('should display error message if package fails to create')
 })
+
+describe('zcliignore', function () {
+  const appPath = path.join(__dirname, 'mocks/single_product_ignore')
+  const tmpPath = path.join(appPath, 'tmp')
+
+  const file = readline.createInterface({
+    input: fs.createReadStream(path.join(appPath, '.zcliignore')),
+    output: process.stdout,
+    terminal: false
+  })
+
+  const ignoreArr: string[] = [] // array that holds each line of the .ignore file
+
+  file.on('line', (line) => {
+    ignoreArr.push(line) // add to array dynamically
+  })
+
+  after(async () => {
+    fs.readdir(tmpPath, (err, files) => {
+      if (err) throw err
+
+      for (const file of files) {
+        fs.unlink(path.join(tmpPath, file), (err) => {
+          if (err) throw err
+        })
+      }
+    })
+  })
+
+  test
+    .env({
+      ZENDESK_SUBDOMAIN: 'z3ntest',
+      ZENDESK_EMAIL: 'admin@z3ntest.com',
+      ZENDESK_PASSWORD: '123456' // the universal password
+    })
+    .nock('https://z3ntest.zendesk.com', api => {
+      api
+        .post('/api/v2/apps/validate')
+        .reply(200)
+    })
+    .stdout()
+    .command(['apps:package', appPath])
+    .it('should not include certain files as specified in .zcliignore', async () => {
+      const packagePath = path.join(tmpPath, (fs.readdirSync(tmpPath).find(fn => fn.startsWith('app'))) + '')
```

Comment:
`fs.readdirSync(tmpPath).find(fn => fn.startsWith('app'))` will return a string so I don't think `+ ''` is necessary here.

Reply:
TS complains about non explicit typing here for me... might just be me though.

Reply:
Aha I got it. Coz `find` may return `undefined` which `path.join` doesn't like so TS complains that.
You can change to the line below
```
const packagePath = path.join(tmpPath, fs.readdirSync(tmpPath).find(fn => fn.startsWith('app')) || '')
```
It means if we `find` it we use it otherwise we use `''`.